### PR TITLE
Update subiquity and adapt to changes

### PR DESCRIPTION
**Pull request by jpnurmi**
- canonical/subiquity#1176
- canonical/subiquity#1181 (`storage/v2/guided` payload)

### CODE REVIEW

**Comment by jpnurmi** [comment]
There are a few more issues while updating to the very latest subiquity. I'm turning this into a draft for further investigation.

**Comment by jpnurmi** [comment]
- no permission to write logs
- apt configurer fails in the CI
- setxkbmap fails in the CI

File: packages/subiquity_client/lib/src/types.dart [hunk 1]
```diff
@@ -255,6 +265,8 @@ class Disk with _$Disk {
     String? vendor,
   }) = _Disk;
 
+  List<Partition>? get partitions => objects?.whereType<Partition>().toList();
```
Comment (jpnurmi) [inline packages/subiquity_client/lib/src/types.dart:268]: Note: The `partitions` list from subiquity includes now both, `Partition` and `Gap` objects, which breaks the UI and a lot of tests. Renaming the mixed object list as `objects` and providing `partitions` for backward compatibility reduces the amount of diff a lot.

Comment (oSoMoN) [reply packages/subiquity_client/lib/src/types.dart:268]: That's very considerate, thanks!

**Review by oSoMoN — APPROVED** [review]
LGTM, thanks!

**Pull request merged**